File: src/data/mockData.ts
export type WeatherForecast = {
  day: string;
  label: string;
  temp: number;
  humidity: number;
  rainChance: number;
  wind: number;
  summary: string;
  icon: string;
};

export type CropProfile = {
  name: string;
  hindi: string;
  growthStages: string[];
  water: string;
  harvest: string;
  note: string;
  icon: string;
};

export type FertilizerSuggestion = {
  name: string;
  description: string;
  why: string;
};

export type FertilizerAdvice = {
  crop: string;
  suggestions: FertilizerSuggestion[];
};

export type DiseaseResult = {
  disease: string;
  confidence: number;
  treatment: string;
  explanation: string;
};

export type FeatureItem = {
  title: string;
  description: string;
  icon: string;
};

export type AwarenessTip = {
  heading: string;
  text: string;
};

export const homeFeatures: FeatureItem[] = [
  {
    title: 'मौसम पूर्वानुमान',
    description: 'कृषि मौसम डेटा को सरल कार्ड में देखें।',
    icon: '☁️',
  },
  {
    title: 'फसल ट्रैकिंग',
    description: 'फसल की वृद्धि और सिंचाई की जानकारी।',
    icon: '🌾',
  },
  {
    title: 'उर्वरक सलाह',
    description: 'फसल के अनुरूप सरल उर्वरक सुझाव।',
    icon: '🌱',
  },
  {
    title: 'AI रोग पहचान',
    description: 'रोग पहचान का डेमो और उपचार सुझाव।',
    icon: '📷',
  },
];

export const weatherForecasts: WeatherForecast[] = [
  {
    day: 'आज',
    label: 'Sunny',
    temp: 32,
    humidity: 54,
    rainChance: 18,
    wind: 12,
    summary: 'हल्की धूप, किसान कार्यों के लिए अनुकूल।',
    icon: '☀️',
  },
  {
    day: 'कल',
    label: 'Cloudy',
    temp: 28,
    humidity: 68,
    rainChance: 42,
    wind: 9,
    summary: 'बादल छाने की संभावना, उचित सिंचाई समय चुनें।',
    icon: '⛅',
  },
  {
    day: 'परसों',
    label: 'Rain',
    temp: 24,
    humidity: 80,
    rainChance: 75,
    wind: 15,
    summary: 'बारिश के आसार, खेत साफ और तैयार रखें।',
    icon: '🌧️',
  },
];

export const cropProfiles: CropProfile[] = [
  {
    name: 'Wheat',
    hindi: 'गेहूं',
    icon: '🌾',
    growthStages: ['बुवाई', 'निकाली', 'हल्की हरियाली', 'कटाई तैयार'],
    water: 'हर 7-10 दिनों में 5-6 लीटर प्रति वर्गमीटर',
    harvest: '120-130 दिन',
    note: 'सर्दियों में सिंचाई योजना पर विशेष ध्यान दें।',
  },
  {
    name: 'Rice',
    hindi: 'धान',
    icon: '🌾',
    growthStages: ['पानी भरा खेत', 'कली फूटना', 'भावन', 'कटाई'],
    water: 'नियमित जल स्तर बनाए रखें, लगभग 8-10 लीटर',
    harvest: '110-140 दिन',
    note: 'गर्मी से पहले सही समय पर जल निकासी करें।',
  },
  {
    name: 'Tomato',
    hindi: 'टमाटर',
    icon: '🍅',
    growthStages: ['रोपण', 'फलन', 'रंग बदलना', 'फसल संग्रह'],
    water: 'हर 3-4 दिन में 3-4 लीटर',
    harvest: '70-90 दिन',
    note: 'फलों पर पानी सीधे न डालें, पत्ती सूखने पर सिंचाई करें।',
  },
  {
    name: 'Potato',
    hindi: 'आलू',
    icon: '🥔',
    growthStages: ['कटाई', 'उद्पत्ति', 'पत्तियाँ फैलना', 'फूस लगना'],
    water: 'हर 5-7 दिन में 4-5 लीटर',
    harvest: '90-110 दिन',
    note: 'पानी अधिक मात्रा में न दें, हल्की नमी पर्याप्त होती है।',
  },
  {
    name: 'Sugarcane',
    hindi: 'गन्ना',
    icon: '🍃',
    growthStages: ['नर्सरी', 'रोपण', 'हरियाली', 'गठना'],
    water: 'सात से दस दिनों में 10-12 लीटर',
    harvest: '10-12 महीने',
    note: 'मिट्टी को नम रखें और खरपतवार मुक्त रखें।',
  },
];

export const fertilizerAdvice: Record<string, FertilizerAdvice> = {
  Wheat: {
    crop: 'Wheat',
    suggestions: [
      {
        name: 'UREA',
        description: 'नाइट्रोजन की आवश्यकता को पूरा करता है, पत्तियों को हरा और स्वस्थ बनाता है।',
        why: 'गेहूं की साइट पर तेजी से विकास और बल बढ़ाने में मदद करता है।',
      },
      {
        name: 'NPK',
        description: 'समग्र पोषण के लिए समतुल्य नाइट्रोजन, फॉस्फोरस और पोटैशियम।',
        why: 'पनपती मजबूत और आहार संतुलन बनाए रखता है।',
      },
      {
        name: 'Organic Compost',
        description: 'मिट्टी की संरचना सुधारता है और दीर्घकालिक पोषण देता है।',
        why: 'रासायनिक उर्वरक पर निर्भरता घटाकर स्वास्थ्य फसलों को बढ़ावा देता है।',
      },
    ],
  },
  Rice: {
    crop: 'Rice',
    suggestions: [
      {
        name: 'NPK',
        description: 'धान की जड़ों के लिए जरूरी पोषक तत्व देता है।',
        why: 'फलन और दाने में वृद्धि में मदद करता है।',
      },
      {
        name: 'UREA',
        description: 'पत्तियों को हरा और लम्बी रखने में मदद करता है।',
        why: 'बोन्साई चरण में लक्षित नाइट्रोजन प्रदान करता है।',
      },
      {
        name: 'Organic Compost',
        description: 'जल धारण क्षमता बढ़ाने और मिट्टी की ताजगी के लिए।',
        why: 'नमी की अवधि के साथ कार्बनिक तत्व डालता है।',
      },
    ],
  },
  Tomato: {
    crop: 'Tomato',
    suggestions: [
      {
        name: 'NPK',
        description: 'फल और फूल विकास के लिए संतुलित पोषण।',
        why: 'मीठे और बड़े टमाटर उत्पादन में मदद करता है।',
      },
      {
        name: 'Organic Compost',
        description: 'मृदा स्वास्थ्य बनाए रखता है और संरचना सुधारता है।',
        why: 'कीट और रोग प्रतिरोधक क्षमता बढ़ाता है।',
      },
    ],
  },
  Potato: {
    crop: 'Potato',
    suggestions: [
      {
        name: 'UREA',
        description: 'तने और पत्तियां मजबूत बनाने के लिए नाइट्रोजन।',
        why: 'उत्पादकता बढ़ाने में मदद करता है।',
      },
      {
        name: 'NPK',
        description: 'आलू बटुए को भरपूर पोषक तत्व देता है।',
        why: 'मानव उपयुक्त मिट्टी और गांठ विकास के लिए।',
      },
      {
        name: 'Organic Compost',
        description: 'मिट्टी में कार्बनिक तत्व और नमी जोड़ता है।',
        why: 'स्वस्थ और स्वादिष्ट फसल बनाता है।',
      },
    ],
  },
  Sugarcane: {
    crop: 'Sugarcane',
    suggestions: [
      {
        name: 'UREA',
        description: 'लंबी अवधि के विकास के लिए नाइट्रोजन समर्थन।',
        why: 'पत्तियों को मजबूत और ऊंचाई बढ़ाने में सहायक।',
      },
      {
        name: 'NPK',
        description: 'कैन क्रॉप में उर्जा और बढ़ती जड़ों के लिए।',
        why: 'काटने योग्य द्रव्यमान और मिठास में सुधार।',
      },
      {
        name: 'Organic Compost',
        description: 'समृद्ध मिट्टी के लिए प्राकृतिक पोषक तत्व।',
        why: 'लोकेशन पर मिट्टी की हालत को स्थिर बनाता है।',
      },
    ],
  },
};

export const diseaseSamples: DiseaseResult[] = [
  {
    disease: 'ब्लाइट रोग',
    confidence: 93,
    treatment: 'संक्रमित पत्तियों को हटाएं, फफूंदनाशक का उपयोग करें।',
    explanation: 'भरी नमी और कमजोर पत्ती होने पर यह रोग जल्दी फैलता है।',
  },
  {
    disease: 'पीला पत्ती रोग',
    confidence: 87,
    treatment: 'ठीक समय पर पानी दें और संतुलित उर्वरक इस्तेमाल करें।',
    explanation: 'जीवाणु जनित रोग होने पर पत्तियों का रंग पीला हो जाता है।',
  },
  {
    disease: 'पत्ती का धब्बेदार रोग',
    confidence: 79,
    treatment: 'रोगग्रस्त भागों को काटें, सूखा मौसम बनाएं।',
    explanation: 'यह रोग आद्रता और असंतुलित पोषण से बढ़ता है।',
  },
];

export const awarenessTips: AwarenessTip[] = [
  {
    heading: 'जल संरक्षण',
    text: 'बारिश का पानी एकत्रित करें और ड्रिप सिंचाई जैसे आधुनिक तरीके अपनाएँ।',
  },
  {
    heading: 'जैविक खेती',
    text: 'रासायनिक उर्वरक के बजाय प्राकृतिक खाद का उपयोग कर मिट्टी की उपज बढ़ाएँ।',
  },
  {
    heading: 'सरकारी योजनाएं',
    text: 'कृषि विभाग की नई योजनाओं के लिए नजदीकी किसान केंद्र से जानकारी लें।',
  },
  {
    heading: 'सदस्यता समूह',
    text: 'छोटे किसान समूह बनाकर लागत साझा करें और अच्छे प्रबंधन निर्णय लें।',
  },
];

File: mobile/lib/data/mock_data.dart
class WeatherForecast {
  final String day;
  final String summary;
  final int temperature;
  final int humidity;
  final int rainChance;
  final int windSpeed;

  WeatherForecast({
    required this.day,
    required this.summary,
    required this.temperature,
    required this.humidity,
    required this.rainChance,
    required this.windSpeed,
  });
}

class CropProfile {
  final String name;
  final String hindi;
  final String icon;
  final List<String> growthStages;
  final String waterNeed;
  final String harvestTime;
  final String note;

  CropProfile({
    required this.name,
    required this.hindi,
    required this.icon,
    required this.growthStages,
    required this.waterNeed,
    required this.harvestTime,
    required this.note,
  });
}

class FertilizerSuggestion {
  final String title;
  final String description;
  final String benefit;

  FertilizerSuggestion({
    required this.title,
    required this.description,
    required this.benefit,
  });
}

class DiseaseResult {
  final String name;
  final int confidence;
  final String treatment;
  final String explanation;

  DiseaseResult({
    required this.name,
    required this.confidence,
    required this.treatment,
    required this.explanation,
  });
}

final weatherForecasts = <WeatherForecast>[
  WeatherForecast(
    day: 'आज',
    summary: 'हल्की धूप, खेत के काम के लिए अच्छा।',
    temperature: 32,
    humidity: 54,
    rainChance: 18,
    windSpeed: 12,
  ),
  WeatherForecast(
    day: 'कल',
    summary: 'बादल छाए रह सकते हैं, सिंचाई को संतुलित रखें।',
    temperature: 28,
    humidity: 68,
    rainChance: 42,
    windSpeed: 9,
  ),
  WeatherForecast(
    day: 'परसों',
    summary: 'बारिश की संभावना, खेत की तैयारी रखें।',
    temperature: 24,
    humidity: 80,
    rainChance: 75,
    windSpeed: 15,
  ),
];

final cropProfiles = <CropProfile>[
  CropProfile(
    name: 'Wheat',
    hindi: 'गेहूं',
    icon: '🌾',
    growthStages: ['बुवाई', 'निकाल', 'हरियाली', 'कटाई'],
    waterNeed: 'हर 7-10 दिन में 5-6 लीटर प्रति वर्गमीटर',
    harvestTime: '120-130 दिन',
    note: 'ठंडी फसल, समय पर गहरी जुताई जरूरी।',
  ),
  CropProfile(
    name: 'Rice',
    hindi: 'धान',
    icon: '🌾',
    growthStages: ['पानी भरा खेत', 'कली फूटना', 'भावन', 'कटाई'],
    waterNeed: '8-10 लीटर नियमित पानी',
    harvestTime: '110-140 दिन',
    note: 'मिट्टी को संतुलित पानी रखें।',
  ),
  CropProfile(
    name: 'Tomato',
    hindi: 'टमाटर',
    icon: '🍅',
    growthStages: ['रोपण', 'फलन', 'रंग बदलना', 'फसल संग्रह'],
    waterNeed: 'हर 3-4 दिन में 3-4 लीटर',
    harvestTime: '70-90 दिन',
    note: 'फल पर पानी न डालें, पत्ती सुखी रखें।',
  ),
  CropProfile(
    name: 'Potato',
    hindi: 'आलू',
    icon: '🥔',
    growthStages: ['कटाई', 'उद्भव', 'पत्तियाँ फैलना', 'फूस लगना'],
    waterNeed: 'हर 5-7 दिन में 4-5 लीटर',
    harvestTime: '90-110 दिन',
    note: 'मृदा का नमी संतुलन बनाए रखें।',
  ),
  CropProfile(
    name: 'Sugarcane',
    hindi: 'गन्ना',
    icon: '🍃',
    growthStages: ['नर्सरी', 'रोपण', 'हरियाली', 'कटाई तैयार'],
    waterNeed: '10-12 लीटर प्रत्येक सप्ताह',
    harvestTime: '10-12 महीने',
    note: 'धीमी वृद्धि वाली बड़ी फसल, निरंतर देखभाल चाहिए।',
  ),
];

final fertilizerAdvice = <String, List<FertilizerSuggestion>>{
  'Wheat': [
    FertilizerSuggestion(
      title: 'UREA',
      description: 'चक्रीय नाइट्रोजन प्रदान करता है और पत्तियों को हरा बनाता है।',
      benefit: 'तेजी से विकास और मजबूत पत्ती संरचना।',
    ),
    FertilizerSuggestion(
      title: 'NPK',
      description: 'समग्र पोषण के लिए संतुलित मिश्रण।',
      benefit: 'बेहतर कटाई और खेत की उपज।',
    ),
    FertilizerSuggestion(
      title: 'Organic Compost',
      description: 'मिट्टी की गुणवत्ता लंबे समय तक बढ़ाता है।',
      benefit: 'स्वस्थ मिट्टी और पर्यावरण समर्थित खेती।',
    ),
  ],
  'Rice': [
    FertilizerSuggestion(
      title: 'NPK',
      description: 'धान की अनुकूल वृद्धि के लिए पुरक।',
      benefit: 'बेहतर दाना और मजबूत जड़ प्रणाली।',
    ),
    FertilizerSuggestion(
      title: 'UREA',
      description: 'पत्तियों को पोषण देता है।',
      benefit: 'हराभरा तना और उत्पादन बढ़ाता है।',
    ),
    FertilizerSuggestion(
      title: 'Organic Compost',
      description: 'जल धारण क्षमता और मिट्टी संरचना सुधारता है।',
      benefit: 'लंबी अवधि तक उपज बनाए रखता है।',
    ),
  ],
  'Tomato': [
    FertilizerSuggestion(
      title: 'NPK',
      description: 'फल और फूल विकास के लिए।',
      benefit: 'टमाटर बड़े और रसदार बनते हैं।',
    ),
    FertilizerSuggestion(
      title: 'Organic Compost',
      description: 'मृदा स्वास्थ्य को प्राकृतिक रूप से बढ़ाता है।',
      benefit: 'बीमारियों की प्रतिरोधक क्षमता।',
    ),
  ],
  'Potato': [
    FertilizerSuggestion(
      title: 'UREA',
      description: 'तने और पत्तियों के विकास में सहायक।',
      benefit: 'उत्पादकता में बढ़ोतरी।',
    ),
    FertilizerSuggestion(
      title: 'NPK',
      description: 'आलू के गांठों को पोषण देता है।',
      benefit: 'गांठे बड़े और भरपूर बनती हैं।',
    ),
    FertilizerSuggestion(
      title: 'Organic Compost',
      description: 'मिट्टी में जैविक पदार्थ जोड़ता है।',
      benefit: 'स्वस्थ उपज और स्वाद।',
    ),
  ],
  'Sugarcane': [
    FertilizerSuggestion(
      title: 'UREA',
      description: 'लंबी उम्र की फसल को नाइट्रोजन देता है।',
      benefit: 'लंबाई और हरियाली बढ़ती है।',
    ),
    FertilizerSuggestion(
      title: 'NPK',
      description: 'मिठास और जड़ विकास में मदद।',
      benefit: 'बेहतर उपज और मजबूत पौधा।',
    ),
    FertilizerSuggestion(
      title: 'Organic Compost',
      description: 'मिट्टी की सतह को समृद्ध करता है।',
      benefit: 'दीर्घकालिक मिट्टी स्वास्थ्य।',
    ),
  ],
};

final diseaseSamples = <DiseaseResult>[
  DiseaseResult(
    name: 'ब्लाइट रोग',
    confidence: 93,
    treatment: 'संक्रमित पत्तियों को हटाएं और फफूंदनाशक का उपयोग करें।',
    explanation: 'आर्द्रता में यह रोग तेजी से फैलता है।',
  ),
  DiseaseResult(
    name: 'पीला पत्ती रोग',
    confidence: 87,
    treatment: 'संतुलित उर्वरक और समय पर सिंचाई करें।',
    explanation: 'पत्ती पीली और कमजोर हो जाती है।',
  ),
  DiseaseResult(
    name: 'धब्बेदार रोग',
    confidence: 79,
    treatment: 'रोगग्रस्त हिस्सों को काटें और खेत को सूखा रखें।',
    explanation: 'गंदगी और नमी से यह रोग बढ़ता है।',
  ),
];

final awarenessTips = <String>[
  'बारिश का पानी एकत्रित करके सिंचाई बचाएं।',
  'जैविक खाद से मिट्टी को स्वस्थ बनाएं।',
  'सरकारी योजनाओं की जानकारी अपने किसान केंद्र से प्राप्त करें।',
  'किसान समूह बनाकर लागत साझा करें।',
];
